src/mon/MonmapMonitor.cc

```diff
@@ -808,3 +830,118 @@ void MonmapMonitor::check_sub(Subscription *sub)
     }
   }
 }
+
+namespace ceph::mon_cmds {
+
+using std::string;
+using std::vector;
+
+using ceph::mon_cmds::mon_cmd;
+using ceph::mon_cmds::mon_cmd_result;
+
+// A demo type for testing hooks:
+#pragma warning JFW REMOVEME
+struct cmd_echo : public mon_cmd 
+{
+ vector<string> tags() { return { "mon echo" }; }
+
+ mon_cmd_result at_prepare(const mon_cmd_ctx& state, string_view cmd)
+ {
+    state.ss << "cmd_echo() at_prepare: " << state.prefix;
+    return mon_cmd_result::success;
+ }
+
+ mon_cmd_result at_preprocess(const mon_cmd_ctx& state, string_view cmd)
+ {
+    state.ss << "cmd_echo() at_preprocess: " << state.prefix;
+    return mon_cmd_result::success;
+ }
+};
+
+struct cmd_getmap : public mon_cmd 
```

Comment:
Here's a port of an actual existing command, which has code responding to both "mon getmap" and "mon dump".